### PR TITLE
Build own TCP framework. Bullet server side creation implemented.

### PR DESCRIPTION
**Pull request by Ploug**

### CODE REVIEW

File: src/application/communication/Broker.java [hunk 1]
```diff
@@ -189,9 +203,12 @@ private void receive()
                         int id = buffer.getInt();
                         double x = buffer.getDouble();
                         double y = buffer.getDouble();
-                        Point2D position = new Point2D(x, y);
+                        GameObjectDAO objectPos = new GameObjectDAO();
```
Comment (LuqJensen) [inline src/application/communication/Broker.java:203]: constructor??

File: src/application/communication/Broker.java [hunk 2]
```diff
@@ -125,20 +136,23 @@ public void sendUpdate(List<ObjectPosition> posList ) // TODO Should possibly kn
...
+        buffer.put((byte)Opcodes.POSITION_UPDATE.getValue());
         buffer.put((byte)2); // ASCII Standard for Start of text
-        for(ObjectPosition pos : posList)
+        for (GameObjectDAO pos : posList)
         {
-            buffer.putInt(pos.Id);
-            buffer.putDouble(pos.Position.getX());
-            buffer.putDouble(pos.Position.getY());
-            buffer.put((byte)31); // Unit   
+            if (buffer.position() > 950)
```
Comment (LuqJensen) [inline src/application/communication/Broker.java]: what?

File: src/application/communication/Broker.java [hunk 3]
```diff
@@ -227,22 +244,52 @@ public void send(byte[] buf)
...
+     * 
+     * @return The id of the bullet created.
+     */
+    public void requestBulletCreation(GameObjectDAO data)
+    {
+        ByteBuffer buffer = ByteBuffer.allocate(256);
+
+        buffer.order(ByteOrder.LITTLE_ENDIAN);
+        buffer.put((byte)1); // ASCII Standard for Start of heading
+        buffer.put((byte)Opcodes.REQUEST_BULLET.getValue());
```
Comment (LuqJensen) [inline src/application/communication/Broker.java:272]: ordinal?

File: src/application/communication/Broker.java [hunk 4]
```diff
@@ -258,18 +305,9 @@ private void startListeningTCP()
...
-                        {
-                            value = input.read();
-                            System.out.println("data: "+value);
-                           //Do something while value isnt end of transmission.
-                        }
-                        while(value != -1); // TODO end of transmission value is?
-                            
-                    }
+                    byte[] buf = new byte[512];
+                    input.read(buf);
```
Comment (LuqJensen) [inline src/application/communication/Broker.java:307]: if (buf.length == 0) break/return

File: src/application/communication/Broker.java [hunk 5]
```diff
@@ -279,4 +317,94 @@ private void startListeningTCP()
...
+     */
+    private void handleBulletCreation(ByteBuffer input)
+    {
+        GameObjectDAO data = new GameObjectDAO();
+        data.x = input.getDouble();
+        data.y = input.getDouble();
+        data.height = input.getDouble();
+        data.velocityX = input.getDouble();
+        data.velocityY = input.getDouble();
+        data.bulletType = Bullet.Type.values()[input.getInt()];
```
Comment (LuqJensen) [inline src/application/communication/Broker.java]: this is very very expensive. instead make a static array in the enum class from which you can get opcode by index

File: src/application/communication/Broker.java (hunk 3 above)
Comment (Ploug) [inline src/application/communication/Broker.java:272]: ordinal returns its offset in the enum, not the specific value that we provie it in the private constructor we've made.

**Pull request merged**